apps/src/turtle/turtle.js

```diff
@@ -407,6 +407,37 @@ class Visualization {
     this.heading = heading;
   }
 
+  reset(startDirection = DEFAULT_DIRECTION, initialX = DEFAULT_X, initialY = DEFAULT_Y) {
+    // Standard starting location and heading of the turtle.
+    this.x = initialX;
+    this.y = initialY;
+    this.heading = startDirection;
+    this.penDownValue = true;
+    this.avatar.visible = true;
+    this.resetTurtleFrame();
+
+    // Clear the display.
+    this.ctxScratch.canvas.width = this.ctxScratch.canvas.width;
+    this.ctxPattern.canvas.width = this.ctxPattern.canvas.width;
+    if (this.isFrozenSkin) {
+      this.ctxScratch.strokeStyle = 'rgb(255,255,255)';
+      this.ctxScratch.fillStyle = 'rgb(255,255,255)';
+      this.ctxScratch.lineWidth = 2;
+    } else {
+      this.ctxScratch.strokeStyle = '#000000';
+      this.ctxScratch.fillStyle = '#000000';
+      this.ctxScratch.lineWidth = 5;
+    }
+
+    this.ctxScratch.lineCap = 'round';
+    this.ctxScratch.font = 'normal 18pt Arial';
+    this.display();
+
+    // Clear the feedback.
+    this.ctxFeedback.clearRect(
+      0, 0, this.ctxFeedback.canvas.width, this.ctxFeedback.canvas.height);
```

Comment:
Should this be indented 4 spaces, since it is a line continuation?

Reply:
Good question!  Our style guide puts continuations at 4 spaces, but both 2 and 4 spaces are common in our repository.  We should probably update the style guide 😄.